dbms/src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -160,8 +167,13 @@ void PushingToViewsBlockOutputStream::process(const Block & block, size_t view_n
 
     try
     {
-        BlockInputStreamPtr from = std::make_shared<OneBlockInputStream>(block);
-        InterpreterSelectQuery select(view.query, *views_context, from);
+        /// We create a table with the same name as original table and the same alias columns,
+        ///  but it will contain single block (that is INSERT-ed into main table).
+        /// InterpreterSelectQuery will do processing of alias columns.
+        Context local_context = *views_context;
+        local_context.addExternalTable(storage->getTableName(), StorageBlock::create(block, storage), nullptr);
+        InterpreterSelectQuery select(view.query, local_context);
```

Comment:
Non obvious, how it is guaranteed that it will read from external table
if full qualified name (db.table) is in `view.query`.